### PR TITLE
hack: workaround torch bug to unblock mixtral fine-tuning

### PR DESCRIPTION
**Pull request by geoffreyangus**
Workaround for https://github.com/huggingface/transformers/issues/28023

Revert when the above issue is closed.

### CODE REVIEW

**Review by arnavgarg1 — APPROVED** [review]
LGTM barring a minor nit on attn_implementation and also potentially considering creating 2 functions that can be re-used.

**Review by tgaddair — APPROVED** [review]
I would just do a quick sanity check to make sure deepspeed works with these changes

File: ludwig/models/llm.py [hunk 1]
```diff
@@ -88,6 +88,7 @@ def load_pretrained_from_config(
         load_kwargs["torch_dtype"] = getattr(torch, config_obj.quantization.bnb_4bit_compute_dtype)
         load_kwargs["quantization_config"] = config_obj.quantization.to_bitsandbytes()
         load_kwargs["device_map"] = "auto"
+        load_kwargs["attn_implementation"] = "eager"
```
[the remark this answers is not in the document]
Comment (geoffreyangus) [reply ludwig/models/llm.py:91]: Seems like it's okay: https://huggingface.co/docs/transformers/main_classes/configuration#transformers.PretrainedConfig.attn_implementation

Didn't break the tests either, so I think we should be good!

[the remark this answers is not in the document]
Comment (arnavgarg1) [reply ludwig/models/llm.py]: Yup! Thanks!

**Review by arnavgarg1 — APPROVED** [review]
LGTM! Thanks for the workaround!

**Review by tgaddair — APPROVED** [review]

**Pull request merged**